theano/gof/vm.py

```diff
@@ -378,6 +379,13 @@ def run_thunk_of_node(self, node):
 
         """
         idx = self.node_idx[node]
+        if self.callback_before is not None:
+            self.callback_before(
```

Comment:
When we discussed this, I was more thinking that callback_before would be called before the function start execution, not before each node. This add extra overhead. I don't see a need for that case of callback before each node.

What do you think of that?